cmd/kyma/sync/function/function.go

```diff
@@ -0,0 +1,101 @@
+package function
+
+import (
+	"context"
+	"github.com/kyma-incubator/hydroform/function/pkg/workspace"
+	"github.com/kyma-project/cli/internal/cli"
+	"github.com/kyma-project/kyma/components/function-controller/pkg/apis/serverless/v1alpha1"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+)
+
+const (
+	defaultNamespace = "default"
+	functions = "functions"
+)
+
+type command struct {
+	opts *Options
+	cli.Command
+}
+
+//NewCmd creates a new init command
+func NewCmd(o *Options) *cobra.Command {
+	c := command{
+		opts:    o,
+		Command: cli.Command{Options: o.Options},
+	}
+	cmd := &cobra.Command{
+		Use:   "function",
+		Short: "Saves locally resources for requested Function.",
+		Long: `Use this command to create the local workspace with the default structure of your Function's code and dependencies. Update this configuration to your references and apply it to a Kyma cluster. 
+Use the flags to specify the initial configuration for your Function or to choose the location for your project.`, //todo
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return c.Run()
+		},
+	}
+
+	cmd.Flags().StringVar(&o.Name, "name", "", `Function name.`)
+	cmd.Flags().StringVarP(&o.Namespace, "namespace","n", defaultNamespace, `Namespace from which you want to sync Function.`)
+	cmd.Flags().StringVarP(&o.OutputPath, "output", "o", "", `Full path to the directory where you want to save the project.`)
+	cmd.Flags().StringVarP(&o.Kubeconfig, "kubeconfig", "k", "", `Full path to the Kubeconfig file.`)
+
+	return cmd
+}
+
+func (c *command) Run() error {
+	s := c.NewStep("Generating project structure")
+
+	if err := c.opts.setDefaults(); err != nil {
+		s.Failure()
+		return err
+	}
+
+	if _, err := os.Stat(c.opts.OutputPath); os.IsNotExist(err) {
+		err = os.MkdirAll(c.opts.OutputPath, 0700)
+		if err != nil {
+			return err
+		}
+	}
+
+
+	crdConfig, err := clientcmd.BuildConfigFromFlags("", c.opts.Kubeconfig)
```

Comment:
What happens here when no kubeconfig is provided? Is then the `KUBECONFIG` env variable used? the default config location (`~/.kube/config`)?

Please use the kubeconfig and k8s client utilities built into the CLI, they have been tested and improved over long time:
https://github.com/kyma-project/cli/blob/bad193c82b5fa538f69335789e64bb536beb2200/internal/kube/client.go#L41